### PR TITLE
Use DefaultValue for default value of InputValueDefinition

### PR DESCRIPTION
**Pull request by tobias-tengler**

### CODE REVIEW

**Comment by tobias-tengler** [comment]
@captbaritone Next PR on the road to SchemaDocument resolution-path support :)

File: compiler/crates/schema/src/in_memory/mod.rs [hunk 1]
```diff
@@ -1675,7 +1675,11 @@ impl InMemorySchema {
                     Ok(Argument {
                         name: ArgumentName(arg_def.name.value),
                         type_: self.build_input_object_reference(&arg_def.type_)?,
-                        default_value: arg_def.default_value.clone(),
+                        default_value: if let Some(default_value) = &arg_def.default_value {
+                            Some(default_value.value.clone())
+                        } else {
+                            None
+                        },
```
Comment (captbaritone) [inline compiler/crates/schema/src/in_memory/mod.rs:1682]: Here (and the similar place in relay docblock) could you use `default_value.map(...)`? Might be a little cleaner.

File: compiler/crates/graphql-syntax/src/node/type_system.rs [hunk 2]
```diff
@@ -515,7 +515,7 @@ impl fmt::Display for DirectiveLocation {
 pub struct InputValueDefinition {
     pub name: Identifier,
     pub type_: TypeAnnotation,
-    pub default_value: Option<ConstantValue>,
+    pub default_value: Option<DefaultValue>,
```
Comment (captbaritone) [inline compiler/crates/graphql-syntax/src/node/type_system.rs:518]: Is there more context on why it's useful to have a DefaultValue here?

If we want to do this, I think we should move where `DefaultValue` is defined, perhaps into `compiler/crates/graphql-syntax/src/node/constant_value.rs`? Right now it's defined with the executable definition AST constructs, so I don't think it makes sense to share it between both types of documents without moving it.

**Review by captbaritone — CHANGES_REQUESTED** [review]

File: compiler/crates/graphql-syntax/src/node/type_system.rs (hunk 2 above)
Comment (tobias-tengler) [reply compiler/crates/graphql-syntax/src/node/type_system.rs:518]: The `ConstantValue` doesn't have span information, so we can't determine whether we are on the default value of a `InputValueDefinition` in the path resolution. The `DefaultValue` has a span definition and made sense to me here, as it's the same concept as a default value for a `VariableDefinition`.

Comment (captbaritone) [reply compiler/crates/graphql-syntax/src/node/type_system.rs:518]: Cool. What do you think about moving the definition into constant_value.rs?

**Pull request closed without merging**